src/program/latex/mod.rs

```diff
@@ -0,0 +1,1291 @@
+    }
+}
+
+/// A Wire represents a single qubit. A wire only needs to keep track of all
+/// the elements it contains mapped to some arbitrary column. Diagram keeps
+/// track of where the Wire belongs in the larger circuit, its row, and knows
+/// how each Wire relates to each other at that column. When Diagram parses the
+/// wires as a collection, if the Wire relates to another at some column, then
+/// its field will be updated at that column based on the knowledge Diagram has
+/// about this connection. This updated value also looks arbitrary to Wire, it
+/// does not explicitly define which qubit it relates to, but a digit that
+/// describes how far away it is from the related qubit based on Quantikz.
+#[derive(Clone, Debug, Default)]
+struct Wire {
+    /// the Gates on the wire callable by the column
+    gates: HashMap<u32, Gate>,
+    /// at this column the wire is a control
+    ctrl: HashMap<u32, i64>,
+    /// at this column is the wire a target?
+    targ: HashMap<u32, bool>,
+    /// the Parameters on the wire callable by the column
+    parameters: HashMap<u32, Vec<Parameter>>,
+    /// the Modifiers on the wire callable by the column
+    modifiers: HashMap<u32, Vec<GateModifier>>,
+    /// empty column
+    empty: HashMap<u32, Command>,
+}
+
+impl Wire {
+    /// Retrieves a gate's parameters from Expression and matches them with its
+    /// symbolic definition which is then stored into wire at the specific
+    /// column.
+    ///
+    /// # Arguments
+    /// `&mut self` - exposes the Wire's parameters at this column
+    /// `expression` - expression from Program to get name of Parameter
+    /// `column` - the column taking the parameters
+    /// `texify` - is texify_numerical_constants setting on?
+    fn set_param(&mut self, expression: &Expression, column: u32, texify: bool) {
+        // get the name of the supported expression
+        let text = match expression {
+            Expression::Address(mr) => mr.name.to_string(),
+            Expression::Number(c) => c.re.to_string(),
+            expression => expression.to_string(),
+        };
+
+        // if texify_numerical_constants
+        let param = if texify {
+            // get the matching symbol from text
+            vec![Parameter::Symbol(Symbol::match_symbol(text))]
+        } else {
+            // set the symbol as text
+            vec![Parameter::Symbol(Symbol::Text(text))]
+        };
+
+        self.parameters.insert(column, param);
+    }
+}
+
+#[derive(Clone, Debug, thiserror::Error)]
+pub enum LatexGenError {
+    #[error("Tried to parse CNOT and found a control qubit without a target.")]
+    FoundCNOTWithNoTarget,
+    #[error("The FORKED modifier is unsupported.")]
+    UnsupportedModifierForked,
+    #[error("This instruction is unsupported: {instruction}.")]
+    UnsupportedInstruction { instruction: String },
+    #[error("This gate is unsupported: {gate}.")]
+    UnsupportedGate { gate: String },
+}
+
+pub trait Latex {
```

Comment:
IMO this should be `ToLatex` (c.f. `ToString`)

Reply:
Also I think it would be very funny if you named it `ToLaTeX`

Reply:
```
trait ToLatex {
    fn to_latex(self, settings: RenderSettings) -> Result<String, LatexGenError>;
}

impl<T> ToLatex for T {
    fn to_latex(self, settings: RenderSettings) -> Result<String, LatexGenError> {
        panic!("actually it's pronounced LaTeX")
    }
}
```

Reply:
Renamed to `ToLatex`. It was previously named this. I think I liked the simplicity of `Latex`, but `ToLatex` is definitely more clear as to the purpose of this feature.